document-store/src/main/java/org/hypertrace/core/documentstore/postgres/PostgresCollection.java

```diff
@@ -270,18 +276,32 @@ protected String parseQueryForCompositeFilter(Filter filter) {
       case AND: {
         String childList =
             Arrays.stream(filter.getChildFilters())
-                .map(this::parseQuery)
+                .map(childFilter -> parseFilter(childFilter, paramsBuilder))
                 .filter(str -> !StringUtils.isEmpty(str))
                 .map(str -> "(" + str + ")")
                 .collect(Collectors.joining(" AND "));
         return !childList.isEmpty() ? childList : null;
       }
       default:
         throw new UnsupportedOperationException(
-            String.format("Boolean operation:%s not supported", op));
+            String.format("Query operation:%s not supported", op));
     }
   }
 
+  @VisibleForTesting
+  protected PreparedStatement buildPreparedStatement(String sqlQuery, Params params) throws SQLException {
+    PreparedStatement preparedStatement = client.prepareStatement(sqlQuery);
+    params.getStringParams().forEach((k, v) -> {
+      try {
+        // Postgres Index starts from 1
+        preparedStatement.setString(k+1, v);
```

Comment:
Since the Params class is specific to the Postgres implementation, should we initialize the default in Params to 1 and directly use k here instead of doing a `k+1` here?

Reply:
Yes, agree on that